node/web/domain/cart/Cart.js

```diff
@@ -0,0 +1,45 @@
+
+module.exports.name = 'Cart';
+module.exports.dependencies = ['Blueprint', 'ObjectID', 'exceptions'];
+module.exports.factory = function (Blueprint, ObjectID, exceptions) {
+    'use strict';
+
+    var Cart = function (cart) {
+
+        var self = {
+            addToCart: undefined,
+            deleteFromCart: undefined
+        };
+        self.totalAmount = cart.totalAmount;
+        self.books = cart.books;
+        self.addToCart = function (book) {
+            var inCart = false;
+            var i;
+            for (i = 0; i < self.books.length; i++) {
+                if (self.books[i].uid === book.uid) {
+                    inCart = true;
+                    self.books[i].amountofBooks = self.books[i].amountofBooks + 1;
+                 
+                } else {
+                       inCart = false;
+                }
+            }
+            if (inCart === false) {
+                book.amountofBooks = 1;
+                self.books.push(book);
+            }
+            self.totalAmount = self.totalAmount + book.price;
```

Comment:
Here, you denormalize the price of the books into a stored field. Under some circumstances, this may be acceptable. Note that, if the price of the books were to change, the user's total would not reflect the changes. For instance, let's say a user adds a book that's on sale to their cart on Tuesday. On Wednesday, it is no longer on sale, but when they go to checkout, they will still receive the sale price. As an alternative, you could provide a function that queries all of the books that are in the cart, and returns the sum of their prices.